src/app/services/push.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+
+import { ApiResponse } from '../models/api-response';
+import { getStatuses, getMetadata } from '../utils';
+@Injectable({
+	providedIn: 'root'
+})
+export class PushService {
+
+	constructor( private http: HttpClient ) { }
+
+	public postData(data: ApiResponse) {
+		const jsonpUrl = 'https://api.loklak.org/api/push.json';
+		const headers = new HttpHeaders({
+			'Content-Type': 'application/x-www-form-urlencoded',
+			'Accept': 'application/json',
+			'cache-control': 'no-cache'
+			});
+		if (data) {
+				const temp = '' + JSON.stringify(data);
+				if (JSON.parse(temp)['aggregations']) {
+					const subTemp = temp.substr(0, temp.length - 19) + '}';
+					if (JSON.parse(subTemp)['statuses'][0]) {
+						const metadata = '{"search_metadata":' + getMetadata(subTemp) + ',"statuses":';
+						const value = metadata + getStatuses(subTemp) + '}';
+						const body = new HttpParams()
+							.set('data', value);
+						this.http.post(jsonpUrl, body.toString(), {headers: headers}).subscribe(val => console.log(val));
```

Comment:
Don't we need to dispatch any actions with this event?

Reply:
We don't need an additional action for this, although we can create one. But what could it be used for? For stating a success on push? We can create one for it, but it would not be used as of now anywhere else. And we can not use already available SEARCH_COMPLETE_SUCCESS which do not favours this context.

Reply:
If I am missing on any point, then please guide me to it.